core-web/libs/block-editor/src/lib/nodes/contentlet-block/contentlet-block.node.ts

```diff
@@ -39,17 +39,35 @@ export const ContentletBlock = (injector: Injector): Node<ContentletBlockOptions
         },
 
         renderHTML({ HTMLAttributes }): DOMOutputSpec {
-            const { data } = HTMLAttributes;
-            const img = data.hasTitleImage ? ['img', { src: data.image }] : ['span', {}];
-
-            return [
-                'div',
-                { 'data-dotCMS-contentlet': 'true' },
-                ['h3', data.title],
-                ['div', data.identifier],
-                img,
-                ['div', {}, data.language]
+            type ContentletData = {
+                title?: string;
+                identifier?: string;
+                language?: string | number;
+                languageId?: string | number;
+                hasTitleImage?: boolean;
+                image?: string;
+            };
+
+            const rawData: ContentletData =
+                (HTMLAttributes as { data?: ContentletData })?.data ?? {};
+
+            const titleText = rawData.title ?? '';
+            const identifierText = rawData.identifier ?? '';
+            const languageText = rawData.language ?? rawData.languageId ?? '';
+            const hasImage = Boolean(rawData.hasTitleImage && rawData.image);
```

Comment:
I think here you need to use OR 
Because the `hasTitleImage` property tells you if `titleImage` is present in the object and that property has an string which value is the name of the property that contains the image

Reply:
What I mean, is that not all the time `titleImage` value will be `image` so this can be failing on those cases and lead to false negatives

Reply:
Done!